content/05-expression.md

```diff
@@ -559,6 +596,21 @@ It is important to note that `final` may not have the intended effect with types
 
 
 
+<!--label:expression-local-static-->
+### Local static variables
+
+##### since Haxe 4.3.0
+
+`static var` and `static final` declarations within a function create variables that are scoped to the function (they cannot be referred to from outside the function), but are only initialized once.
+
+This can be used to implement a cache or memoization pattern for functions without creating too many class fields.
+
+[code asset](assets/LocalStatic.hx)
+
+Similarly to [static class variables](class-field-static), the initial value assigned to a local static variable (if any) must not refer to instance variables or arguments of the current function.
```

Comment:
Something to clarify here is that there are restrictions for the initialization of these local statics. In particular, the code cannot refer to `this`.

Reply:
btw I had some issues trying to use it (mostly with macros..), things like `@:persistent` don't seem possible (metadata as a whole?) and initializing to a `new Map` was broken very easily (compilation server issue?).
Will try to find some time early next week to explore the failing cases if you want, maybe that'd be 4.3.1 material

Reply:
Oh wait, I think I commented on the wrong line because that line says exactly what I complained about...